### DISCUSSION TITLE
The plain HTTP request was sent to HTTPS port

### DISCUSSION THREAD

**Discussion by volkorny** [istio/istio#52493] [Q&A] [answered]
Hello,

I want to configure our mesh to act as a proxy (with host header replacement) for external clients. Requests from external clients should go to https://project1.example.com/, be directed to the Istio public ingress, and then be transferred inside the mesh to the external service at https://project2.example.com/ (with the host header set to project2.example.com).

I have done some research and configuration, but now I am encountering a problem. I receive a "400 Bad Request" response with the message: "The plain HTTP request was sent to HTTPS port." This error appears both in the response and in the logs of the project2.example.com server.

Please check my configuration and help me resolve this issue. Thanks!

P.S. The host header is also currently not being replaced. I am working on it with auto_host_rewrite.

```
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: project1-example-com
  namespace: istio-ingress
spec:
  dnsNames:
  - project1.example.com
  issuerRef:
    kind: ClusterIssuer
    name: cert-manager-webhook-dnsimple-production
  secretName: project1-example-com-tls
---
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  annotations:
    external-dns.alpha.kubernetes.io/controller: ignored-resource
  name: project1-gateway
  namespace: test-mesh
spec:
  selector:
    istio: ingress-public-gateway
  servers:
  - hosts:
    - project1.example.com
    port:
      name: http
      number: 80
      protocol: HTTP
    tls:
      httpsRedirect: true
  - hosts:
    - project1.example.com
    port:
      name: https
      number: 443
      protocol: HTTPS
    tls:
      credentialName: istio-ingress/project1-example-com-tls
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: project1-virtualservice
  namespace: test-mesh
  annotations:
    external-dns.alpha.kubernetes.io/hostname: "project1.example.com"
    external-dns.alpha.kubernetes.io/ttl: "60"
spec:
  hosts:
  - "*"
  gateways:
  - test-mesh/project1-gateway
  http:
  - match:
    - uri:
        prefix: /
    route:
    - destination:
        host: project2.example.com
        port:
          number: 443
---
apiVersion: networking.istio.io/v1beta1
kind: ServiceEntry
metadata:
  name: egress-project2-example-com
  namespace: test-mesh
spec:
  hosts:
    - project2.example.com
  ports:
    - name: https
      number: 443
      protocol: TLS
  location: MESH_EXTERNAL
  resolution: DNS

**Comment by howardjohn** [comment, marked as answer]
The gateway is configured to terminate the TLS. Noa it's plaintext and you forward it. 

What you need is a Destination Rule to add TLS back before it's sent out

**Reply by howardjohn** [reply]
Note the service entry protocol doesn't say "add TLS", just that the port serves TLS (only really important for sidecar use case)

**Comment by volkorny** [comment]
Thanks @howardjohn for the answer. Problem with port and protocol is solved now.
Please help with host rewrite. I try this but it doesn't work.
Server with project2.example.com receives request with project1.example.com as host, 
I need "project2.example.com" in host header.
I am not sure should it contain
"project1.example.com" or "project2.example.com" in routeConfiguration:vhost:name

```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: auto-host-rewrite
  namespace: test-mesh
spec:
  workloadSelector:
    labels:
      istio: ingress-public-gateway
  configPatches:
  - applyTo: NETWORK_FILTER
    match:
      context: GATEWAY
      listener:
        filterChain:
          filter:
            name: "envoy.filters.network.http_connection_manager"
    patch:
      operation: MERGE
      value:
        typed_config:
          "@type": "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager"
          use_remote_address: true
  - applyTo: HTTP_ROUTE
    match:
      context: GATEWAY
      routeConfiguration:
        vhost:
          name: "project1.example.com:443"
          route:
            action: ROUTE
    patch:
      operation: MERGE
      value:
        route:
          auto_host_rewrite: true

**Reply by howardjohn** [reply]
Sorry missed that part. Youll just want a rewrite.authority in the VS: https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPRewrite

**Reply by volkorny** [reply]
@howardjohn You didn't just make my day; you made my entire vacation, which starts tomorrow!
